homeassistant/components/motion_blinds/cover.py

```diff
@@ -160,6 +163,13 @@ def name(self):
     @property
     def available(self):
         """Return True if entity is available."""
+        if not self._blind.available:
```

Comment:
Rather than doing this as a side effect of reading a property, I believe a best practice is to do this at the time of update.  One example I found was here: https://github.com/home-assistant/core/blob/c7b9a0715d2a5ae1d6f7c2752da3d304fa763131/homeassistant/components/evohome/climate.py#L294 where the entity overrides async_update, then detects that the entity is unavailable. My impression is that something like that might work here too.